agent/src/beerocks/monitor/monitor_thread.h

```diff
@@ -54,6 +54,18 @@ class monitor_thread : public beerocks::socket_thread {
     virtual std::string print_cmdu_types(const beerocks::message::sUdsHeader *cmdu_header) override;
 
 private:
+    /**
+     * @brief mutex to protect the list of STAs being monitored
+     */
+    std::mutex monitoring_sta_mutex;
```

Comment:
What's that for?
Monitor runs in a single context..

Reply:
Just a habit to add mutexes. But since it is a single thread, I will remove it.